kubectl-cloudflow/cmd/deploy.go

```diff
@@ -81,32 +105,198 @@ the stored credentials.
+				util.LogAndExit("Please specify either the blueprint file or the image path")
+			}
+			if cmd.Flag("image").Changed == true {
+				fmt.Printf("Deploying image %s\n", cmd.Flag("image").Value.String())
+			} else if cmd.Flag("blueprint").Changed == true {
+				fmt.Printf("Deploying blueprint %s\n", cmd.Flag("blueprint").Value.String())
+			}
+		},
+		Run: deployOpts.deployImpl,
 	}
+	deployOpts.cmd.Flags().StringVarP(&deployOpts.blueprintFilePath, "blueprint", "b", "", "blueprint file path.")
+	deployOpts.cmd.Flags().StringVarP(&deployOpts.imagePath, "image", "i", "", "docker image path.")
 	deployOpts.cmd.Flags().StringVarP(&deployOpts.username, "username", "u", "", "docker registry username.")
 	deployOpts.cmd.Flags().StringVarP(&deployOpts.password, "password", "p", "", "docker registry password.")
 	deployOpts.cmd.Flags().BoolVarP(&deployOpts.passwordStdin, "password-stdin", "", false, "Take the password from stdin")
 	deployOpts.cmd.Flags().StringArrayVar(&deployOpts.volumeMounts, "volume-mount", []string{}, "Accepts a key/value pair separated by an equal sign. The key should be the name of the volume mount, specified as '[streamlet-name].[volume-mount-name]'. The value should be the name of an existing persistent volume claim.")
+	deployOpts.cmd.Flags().StringToIntVar(&deployOpts.replicas, "scale", map[string]int{}, "Accepts key/value pairs for replicas per streamlet")
 
 	rootCmd.AddCommand(deployOpts.cmd)
 }
 
 func (opts *deployOptions) deployImpl(cmd *cobra.Command, args []string) {
 	version.FailOnProtocolVersionMismatch()
 
-	imageRef := args[0]
+	if (cmd.Flag("image").Changed == true) && (cmd.Flag("blueprint").Changed == true) {
+		util.LogAndExit("%s", "Cannot specify both blueprint and image in command")
+	}
+	if (cmd.Flag("image").Changed == false) && (cmd.Flag("blueprint").Changed == false) {
+		util.LogAndExit("%s", "Need to specify either blueprint or image in command")
+	}
+	// if (cmd.Flag("image").Changed == true) && (cmd.Flag("scale").Changed == true) {
+	// util.LogAndExit("%s", "scale is supported only with blueprint based deployment")
+	// }
+	if cmd.Flag("image").Changed == true {
+		deployImage(cmd, opts, args)
+	} else {
+		deployBlueprint(cmd, opts, args)
+	}
+}
+
+func deployBlueprint(cmd *cobra.Command, opts *deployOptions, args []string) {
+	// read blueprint file and load its content
+	blueprintFile := cmd.Flag("blueprint").Value.String()
+	content, err := ioutil.ReadFile(blueprintFile)
+	if err != nil {
+		util.LogAndExit("%s", err.Error())
+	}
+	config, err := loadBlueprintConfigContent(string(content[:]))
+	if err != nil {
+		util.LogAndExit("%s", err.Error())
+	}
+	fmt.Printf("%s\n", config)
+
+	// get the images from the blueprint
+	images := config.GetNode("blueprint.images").GetObject().Items()
+
+	var imageRefs []deploy.ImageReference
+	var dockerRegistryURL string
+
+	for _, imageRef := range images {
+		ref, err := parseImageReference(imageRef.GetString())
```

Comment:
this is a "foo=docker.io/bla/foo:1234" string right? that's not really a Image Reference?

Reply:
yes, this is a single record from the `images` section of the blueprint. e.g.

```
images {
    aggr = "eu.gcr.io/bubbly-observer-178213/spark-aggregation:184-a48ddfe-dirty"
    ings = "eu.gcr.io/bubbly-observer-178213/akka-cdr-ingestor:184-a48ddfe-dirty"
    outp = "eu.gcr.io/bubbly-observer-178213/akka-java-aggregation-output:184-a48ddfe-dirty"
  }
```

Each of these refers to an image. Suggestion for a better name ?

Reply:
This is fine, you only use what is on the right side of equals sign, which are image references.

Reply:
changed the name of the variable as we discussed yesterday